Tools/MiniBrowser/win/WebKitBrowserWindow.cpp

```diff
@@ -573,6 +578,65 @@ bool WebKitBrowserWindow::canTrustServerCertificate(WKProtectionSpaceRef protect
     return false;
 }
 
+void WebKitBrowserWindow::navigationActionDidBecomeDownload(WKPageRef, WKNavigationActionRef, WKDownloadRef download, const void* clientInfo)
+{
+    setDownloadClient(download, clientInfo);
+}
+
+void WebKitBrowserWindow::navigationResponseDidBecomeDownload(WKPageRef, WKNavigationResponseRef, WKDownloadRef download, const void* clientInfo)
+{
+    setDownloadClient(download, clientInfo);
+}
+
+void WebKitBrowserWindow::contextMenuDidCreateDownload(WKPageRef, WKDownloadRef download, const void* clientInfo)
+{
+    setDownloadClient(download, clientInfo);
+}
+
+void WebKitBrowserWindow::setDownloadClient(WKDownloadRef download, const void* clientInfo)
+{
+    WKDownloadClientV0 client = { };
+    client.base = { 0, clientInfo };
+    client.decideDestinationWithResponse = downloadDecideDestinationWithResponse;
+    client.didFinish = downloadDidFinish;
+    client.didFailWithError = downloadDidFailWithError;
+
+    WKDownloadSetClient(download, &client.base);
+}
+
+WKStringRef WebKitBrowserWindow::downloadDecideDestinationWithResponse(WKDownloadRef, WKURLResponseRef response, WKStringRef suggestedFilename, const void*)
+{
+    auto filename = createString(suggestedFilename);
+    if (filename.empty()) {
+        auto urlRef = adoptWK(WKURLResponseCopyURL(response));
+        auto hostRef = adoptWK(WKURLCopyHostName(urlRef.get()));
+        filename = createString(hostRef.get());
+    }
+
+    std::wstring folderPath;
+    if (!getKnownFolderPath(FOLDERID_Downloads, folderPath))
+        return nullptr;
```

Comment:
Does any message boxes appear in this case? What does it say?

Reply:
No error message is displayed in this case (PolicyAction::Ignore).

Reply:
Let's add `MessageBox` or a release assert here.

Reply:
Fixed.

Reply:
<img width="215" alt="ErrorDialog-1" src="https://github.com/WebKit/WebKit/assets/113365892/22416ee4-bb80-4ce8-bcc6-9e1ed15834c4">